superset/commands/key_value/create.py

```diff
@@ -98,6 +96,7 @@ def create(self) -> Key:
                     entry.id = self.key
             except ValueError as ex:
                 raise KeyValueCreateFailedError() from ex
+
         db.session.add(entry)
         db.session.flush()
```

Comment:
@michael-s-molina what's your thinking about flushing? Sadly there's no DAO used here and thus it's required given than on the next line the `entry.id` is used, but in general should the DAO flush or should it be left up to the caller who is context aware?

Reply:
@john-bodley @michael-s-molina should we create a DAO for the Key Value entities, and leave permalinks et al as the commands? This could clarify things. I can do that refactor if needed.

Reply:
> @michael-s-molina what's your thinking about flushing? Sadly there's no DAO used here and thus it's required given than on the next line the entry.id is used, but in general should the DAO flush or should it be left up to the caller who is context aware?

There's no right or wrong answer but I prefer to execute flush operations only when necessary to minimize database load. That means that the command is responsible to call `flush` when necessary.

> @john-bodley @michael-s-molina should we create a DAO for the Key Value entities, and leave permalinks et al as the commands? This could clarify things. I can do that refactor if needed.

This would definitely improve things and make the code more similar to rest of the application.

Reply:
@john-bodley @michael-s-molina here's a PR to convert the KV commands into a DAO: #29344